src/msw/joystick.cpp

```diff
@@ -329,16 +329,23 @@ wxString wxJoystick::GetProductName() const
     if (joyGetDevCaps(m_joystick, &joyCaps, sizeof(joyCaps)) != JOYERR_NOERROR)
         return wxEmptyString;
 
+
     wxRegKey key1(wxString::Format(wxT("HKEY_LOCAL_MACHINE\\%s\\%s\\%s"),
                    REGSTR_PATH_JOYCONFIG, joyCaps.szRegKey, REGSTR_KEY_JOYCURR));
 
-    key1.QueryValue(wxString::Format(wxT("Joystick%d%s"),
-                                     m_joystick + 1, REGSTR_VAL_JOYOEMNAME),
-                    str);
+    if (key1.Exists()) {
+        key1.QueryValue(wxString::Format(wxT("Joystick%d%s"),
+                                        m_joystick + 1, REGSTR_VAL_JOYOEMNAME),
+                        str);
+    }
 
-    wxRegKey key2(wxString::Format(wxT("HKEY_LOCAL_MACHINE\\%s\\%s"),
-                                        REGSTR_PATH_JOYOEM, str.c_str()));
-    key2.QueryValue(REGSTR_VAL_JOYOEMNAME, str);
+    if (!str.IsEmpty()) {
+        wxRegKey key2(wxString::Format(wxT("HKEY_LOCAL_MACHINE\\%s\\%s"),
+                                    REGSTR_PATH_JOYOEM, str.c_str()));
```

Comment:
`.c_str()` is unnecessary, wxStrings can be passed directly to wxWidgets pseudo-vararg functions.

Reply:
I never hit this code with my device, and consequently can't test it, so I'll leave this one to you.